src/js/flows/deleteSpaces.ts

```diff
@@ -0,0 +1,32 @@
+import {Thunk} from "../state/types"
+import refreshSpaceNames from "./refreshSpaceNames"
+import {getZealot} from "./getZealot"
+import Current from "../state/Current"
+import Investigation from "../state/Investigation"
+
+const deleteSpaces = (ids: string[]): Thunk => (
+  dispatch,
+  getState,
+  {globalDispatch}
+) => {
+  const zealot = dispatch(getZealot())
+  const clusterId = Current.getConnectionId(getState())
+  return Promise.all(
```

Comment:
It'd be awesome if this could just be.

```js
Promise.all(ids.map(id => dispatch(deleteSpace(id))))
  .catch(e => handle(e))
  .finally(() => dispatch(refreshSpaceNames()))
```